pyproject.toml

```diff
@@ -18,6 +18,9 @@ classifiers = [
 dependencies = [
     "PyYAML==6.*",
     "websocket-client==1.*",
+    # Wide version range to preempt conflicts when charms pin a version.
+    "opentelemetry-api~=1.0",
+    "importlib-metadata",
```

Comment:
@dimaqq is it possible these two dependencies could be optional-dependencies?

I have a situation where i have had to pin `ops==2.20.0` in order to avoid dragging in these deps to prevent opentelemetry-api and `importlib-metadata` from installing.

Reply:
its my understanding this is not a required feature of the ops charms, but rather an "optional" addition to the ops charm environment.